crates/hir/src/lib.rs

```diff
@@ -3258,10 +3454,13 @@ impl Impl {
                     .filter(filter),
             )
         });
+
         for id in def_crates
             .iter()
             .flat_map(|&id| Crate { id }.transitive_reverse_dependencies(db))
             .map(|Crate { id }| id)
+            .chain(Crate { id: env.krate }.dependencies(db).into_iter().map(|it| it.krate.id))
+            .chain(std::iter::once(env.krate))
             .chain(def_crates.iter().copied())
```

Comment:
This line is actually not needed, as its included through transitive_reverse_dependencies
```suggestion
```

Reply:
And I think the `unique` call afterwards won't be necessary either anymore then. I think `transitive_reverse_dependencies` already upholds that property?

Reply:
According to tests `unique` is not needed so I guess I'll remove that as well